src/Microsoft.ML.FastTree/Representation/TreeRegressor.cs

```diff
@@ -0,0 +1,169 @@
+﻿using System;
+using System.Linq;
+using Microsoft.ML.Trainers.FastTree.Internal;
+
+namespace Microsoft.ML.FastTree.Representation
+{
+    /// <summary>
+    /// A container class for exposing <see cref="RegressionTree"/>'s attributes to users.
+    /// This class should not be mutable, so it contains a lot of read-only members.
+    /// </summary>
+    public class TreeRegressor
```

Comment:
>TreeRegressor [](start = 17, length = 13)

As already covered, this is *not* a tree regressor for the simple reason that it does not perform regression. It is the *result* of performing regression. The name `RegressionTree` was appropriate. Please rename the internal class something else appropriate.